src/main/java/seedu/address/model/person/Name.java

```diff
@@ -35,7 +35,8 @@ public Name(String name) {
      * Returns true if a given string is a valid name.
      */
     public static boolean isValidName(String test) {
-        return test.matches(VALIDATION_REGEX);
+        String[] nameArray = test.split("");
+        return ((nameArray.length < 40) && test.matches(VALIDATION_REGEX));
```

Comment:
Good implementation of the isValidName to check for name length of less than equal to  40 as proposed in our meetings.